From 4546c2bbc8284fb32279ef5d02c3bf692ad9de7a Mon Sep 17 00:00:00 2001
Message-ID: <4546c2bbc8284fb32279ef5d02c3bf692ad9de7a.1710967273.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1710967273.git.sharlatanus@gmail.com>
References: <cover.1710967273.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 11 Mar 2024 14:13:04 +0000
Subject: [PATCH 12/49] gnu: python-distributed: Speed up tests.

* gnu/packages/python-science.scm (python-distributed) [arguments]
<#:test-flags>: Add option to run tests in parallel. Disable 3 more
flaky tests.
[native-inputs]: Add python-pytest-xdist. Sort alphabetically.

Change-Id: I588c1a1dc82e3208cc1eeeefbdc58fb080775ac0
---
 gnu/packages/python-science.scm | 14 ++++++++++----
 1 file changed, 10 insertions(+), 4 deletions(-)

diff --git a/gnu/packages/python-science.scm b/gnu/packages/python-science.scm
index b6a116f16b..a01c68cbc3 100644
--- a/gnu/packages/python-science.scm
+++ b/gnu/packages/python-science.scm
@@ -23,6 +23,7 @@
 ;;; Copyright © 2022 Antero Mejr <antero@mailbox.org>
 ;;; Copyright © 2022 jgart <jgart@dismail.de>
 ;;; Copyright © 2023, 2024 Troy Figiel <troy@troyfigiel.com>
+;;; Copyright © 2024 Sharlatan Hellseher <sharlatanus@gmail.com>
 ;;;
 ;;; This file is part of GNU Guix.
 ;;;
@@ -1688,7 +1689,8 @@ (define-public python-distributed
     (arguments
      (list
       #:test-flags
-      '(list "-x" "-m"
+      '(list "-n" "auto"
+             "-x" "-m"
              (string-append "not slow"
                             " and not flaky"
                             " and not gpu"
@@ -1814,7 +1816,10 @@ (define-public python-distributed
               ;; These tests are rather flaky
               " and not test_quiet_quit_when_cluster_leaves"
               " and not multiple_clients_restart"
-              " and not test_steal_twice"))
+              " and not test_steal_twice"
+              " and not test_task_groups_update_start_stop"
+              " and not test_web_preload"
+              " and not test_web_preload_worker"))
       #:phases
       #~(modify-phases %standard-phases
           (add-after 'unpack 'versioneer
@@ -1889,10 +1894,11 @@ (define-public python-distributed
            python-urllib3
            python-zict))
     (native-inputs
-     (list python-importlib-metadata
+     (list python-flaky
+           python-importlib-metadata
            python-pytest
            python-pytest-timeout
-           python-flaky
+           python-pytest-xdist
            python-versioneer))
     (home-page "https://distributed.dask.org")
     (synopsis "Distributed scheduler for Dask")
-- 
2.41.0

